src/server/mod.rs

```diff
@@ -53,11 +63,230 @@ where
     F: Fn(Request) -> Fut,
     Fut: Future<Output = http_types::Result<Response>>,
 {
+    let mut out_io = io.clone();
+    
     loop {
+        let mut _io = io.clone();
+        // io for headers and normal body reading
+        let (io_primary, mut io_primary_send) = channel_bufread::BufReader::with_capacity(8 * 1024, io.clone());
+        // io used with Body::io
+        let (io_upgrade, mut io_upgrade_send) = channel_bufread::BufReader::with_capacity(8 * 1024, io.clone());
+        
+        let (io_complete_send, io_complete_recv) = oneshot::channel::<()>();
+        
+        use futures_util::sink::SinkExt;
+        async_std::task::spawn(async move {
+            let mut io_end_fut = io_complete_recv.fuse();
+            
+            let (mut body_start, mut n) = {
+                let mut buf: Vec<u8> = Vec::new();
+                buf.resize(8 * 1024, 0);
```

Comment:
Why not use `with_capacity`?
```suggestion
                let mut buf: Vec<u8> = Vec::with_capacity(8 * 1024);
```

Reply:
with_capacity makes sense, I'm not sure what the tweak you're referring to is.

Reply:
Well, `with_capacity` only reserve the capacity but it does not fill it up. I think you might want this instead, I am not sure if the `_u8` can be elided.
```suggestion
                let mut buf = [0_u8].repeat(8 * 1024);
```

Reply:
It looks like repeat might be more optimized here. That should work fine either way. It's worth noting that I haven't benchmarked this code at all so I have no idea if it even performs decently in the first place.